app/controllers/events_controller.rb

```diff
@@ -12,15 +12,22 @@ def current_user
   def index
     @events = case params[:filter]
               when 'user'
-                User.find(params[:user_id]).events.where('end_time > ?', Time.now)
+                User.find(params[:user_id]).events.where('end_time > ?', Time.now).order(:start_time)
               when 'user_past'
-                User.find(params[:user_id]).events.where('end_time <= ?', Time.now)
+                User.find(params[:user_id]).events.where('end_time <= ?', Time.now).order(:start_time)
               when 'past'
-                Event.where('end_time <= ?', Time.now)
+                Event.where('end_time <= ?', Time.now).order(:start_time)
+              when 'nearest'
+                location = if !request.remote_ip || request.remote_ip == '127.0.0.1'
+                             'Kiev, Ukraine'
+                           else
+                             request.safe_location
+                           end
+                Event.where(venue: Venue.near(location, 900_000, order: 'distance').map(&:id))
```

Comment:
Not sure it's working.
https://apidock.com/rails/ActiveRecord/SpawnMethods/merge
Also, use pluck instead of map (https://apidock.com/rails/ActiveRecord/Calculations/pluck)